pkg/compute/errors.go

```diff
@@ -0,0 +1,33 @@
+package compute
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+// ErrExecTimeout is an error that is returned when an execution times out.
+type ErrExecTimeout struct {
+	ExecutionTimeout time.Duration
+}
+
+func NewErrExecTimeout(executionTimeout time.Duration) ErrExecTimeout {
+	return ErrExecTimeout{
+		ExecutionTimeout: executionTimeout,
+	}
+}
+
+func (e ErrExecTimeout) Error() string {
+	return fmt.Sprintf("Execution timed out after %s", e.ExecutionTimeout)
+}
+
+func (e ErrExecTimeout) Retryable() bool {
+	return true
+}
+
+func (e ErrExecTimeout) Details() map[string]string {
+	return map[string]string{
+		models.DetailsKeyHint: "Try increasing the task timeout or reducing the task size",
```

Comment:
maybe say "increasing task resources" since it's not clear what it means to reduce the size of a task, least to me its not.

Reply:
addressed in https://github.com/bacalhau-project/bacalhau/pull/4151/commits/94c4824dae9dd31fc54c966d0d25e8e32b3014d5